homeassistant/components/atag/__init__.py

```diff
@@ -0,0 +1,251 @@
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+            ATTR_ICON: ICONS[TEMP_CELSIUS],
+        },
+        {
+            ATTR_NAME: "CH Return Temperature",
+            ATTR_ID: "ch_return_temp",
+            ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+            ATTR_ICON: ICONS[TEMP_CELSIUS],
+        },
+        {
+            ATTR_NAME: "Burning Hours",
+            ATTR_ID: "burning_hours",
+            ATTR_UNIT_OF_MEASUREMENT: HOUR,
+            ATTR_DEVICE_CLASS: None,
+            ATTR_ICON: ICONS[FIRE],
+        },
+        {
+            ATTR_NAME: "Flame",
+            ATTR_ID: "rel_mod_level",
+            ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE,
+            ATTR_DEVICE_CLASS: None,
+            ATTR_ICON: ICONS[FIRE],
+        },
+    ],
+    CLIMATE: {ATTR_NAME: CLIMATE.title()},
+    WATER_HEATER: {ATTR_NAME: "Domestic Hot Water"},
+}
+
+
+async def async_setup(hass: HomeAssistant, config):
+    """Set up the Atag component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Atag integration from a config entry."""
+    session = async_get_clientsession(hass)
+    atag = AtagDataStore(session, paired=True, **entry.data)
+
+    try:
+        await atag.async_update()
+        if not atag.sensordata:
+            raise ConfigEntryNotReady
+    except AtagException:
+        raise ConfigEntryNotReady
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = atag
+
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, platform)
+        )
+
+    async def refresh(event_time):
+        """Poll Atag for latest data."""
+        await atag.async_update()
+        dispatcher.async_dispatcher_send(hass, SIGNAL_UPDATE_ATAG)
+
+    async def async_close_session(event):
+        """Close the session on shutdown."""
+        await atag.async_close()
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_close_session)
+
+    hass.data.setdefault(DATA_LISTENER, {})[entry.entry_id] = async_track_time_interval(
+        hass, refresh, SCAN_INTERVAL
+    )
+
+    return True
+
+
+async def async_unload_entry(hass, entry):
+    """Unload Atag config entry."""
+    remove_listener = hass.data[DATA_LISTENER].pop(entry.entry_id)
+    remove_listener()
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+    return unload_ok
+
+
+class AtagEntity(Entity):
+    """Defines a base Atag entity."""
+
+    def __init__(self, atag: AtagDataStore, atag_type: dict) -> None:
+        """Initialize the Atag entity."""
+        self.atag = atag
+
+        self._id = atag_type.get(ATTR_ID)
```

Comment:
```suggestion
        self._id = atag_type[ATTR_ID]
```
This must be defined right?